crates/p2p/src/swarm/mod.rs

```diff
@@ -536,58 +547,233 @@ where
         Ok(())
     }
 
-    #[instrument(skip(self, _event))]
-    async fn handle_response_request_event(
+    #[instrument(skip(self, event))]
+    async fn handle_request_response_event(
         &mut self,
-        _event: RequestResponseEvent<GetMessageRequest, GetMessageResponse>,
+        event: RequestResponseEvent<GetMessageRequest, GetMessageResponse>,
     ) -> P2PResult<()> {
-        // let RequestResponseEvent::Message { peer, message } = event else {
-        //     return Ok(());
-        // };
-
-        // match message {
-        //     request_response::Message::Request {
-        //         request_id,
-        //         request,
-        //         channel,
-        //     } => {
-        //         let Some(request) = typed::GetMessageRequest::from_msg(request) else {
-        //             debug!(%peer, "Peer sent invalid get message request, disconnecting it");
-        //             let _ = self.swarm.disconnect_peer_id(peer);
-        //             return Ok(());
-        //         };
-        //         match request {
-        //             typed::GetMessageRequest::Genesis { operator_id } => todo!(),
-        //             typed::GetMessageRequest::ExchangeSession {
-        //                 scope,
-        //                 operator_id,
-        //                 kind,
-        //             } => {
-        //                 match kind {
-        //                     typed::GetMessageRequestExchangeKind::Setup => self
-        //                         .db
-        //                         .get_deposit_setup(operator_id, scope)
-        //                         .map(|| GetMessageResponse {
-        //                             body: Some(get_message_response::Body::Setup(
-        //                                 DepositSetupExchange {
-        //                                     scope: todo!(),
-        //                                     payload: todo!(),
-        //                                 },
-        //                             )),
-        //                         }),
-        //                     typed::GetMessageRequestExchangeKind::Nonces => self.db.get,
-        //                     typed::GetMessageRequestExchangeKind::Signatures => todo!(),
-        //                 };
-        //             }
-        //         };
-        //     }
-        //     request_response::Message::Response {
-        //         request_id,
-        //         response,
-        //     } => todo!(),
-        // };
-
-        todo!()
+        if let RequestResponseEvent::InboundFailure {
+            peer,
+            request_id,
+            error,
+        } = event
+        {
+            debug!(peer=?peer, error=?error, request_id=?request_id, "Failed to send response");
```

Comment:
Don't use `Debug` when `Display` is available, aslo there is shorter syntax for that:

```suggestion
            debug!(%peer, %error, %request_id, "Failed to send response");
```